pynumaflow/function/multiproc_server.py

```diff
@@ -0,0 +1,260 @@
+
+    Args:
+        map_handler: Function callable following the type signature of UDFMapCallable
+        mapt_handler: Function callable following the type signature of UDFMapTCallable
+        reduce_handler: Function callable following the type signature of UDFReduceCallable
+        sock_path: Path to the UNIX Domain Socket
+        max_message_size: The max message size in bytes the server can receive and send
+        max_threads: The max number of threads to be spawned;
+                     defaults to number of processors x4
+
+    Example invocation:
+    >>> from typing import Iterator
+    >>> from pynumaflow.function import Messages, Message, MessageTs, MessageT, \
+    ...     Datum, Metadata, MultiProcServer
+    ...
+    >>> def map_handler(key: [str], datum: Datum) -> Messages:
+    ...   val = datum.value
+    ...   _ = datum.event_time
+    ...   _ = datum.watermark
+    ...   messages = Messages(Message(val, keys=keys))
+    ...   return messages
+    ...
+    ... def reduce_handler(key: str, datums: Iterator[Datum], md: Metadata) -> Messages:
+    ...           "Not supported"
+    ...
+    >>> def mapt_handler(key: [str], datum: Datum) -> MessageTs:
+    ...   val = datum.value
+    ...   new_event_time = datetime.time()
+    ...   _ = datum.watermark
+    ...   message_t_s = MessageTs(MessageT(val, event_time=new_event_time, keys=key))
+    ...   return message_t_s
+    ...
+    ...
+    >>> grpc_server = MultiProcServer(
+    ...   mapt_handler=mapt_handler,
+    ...   map_handler=map_handler,
+    ... )
+    >>> grpc_server.start()
+    """
+
+    def __init__(
+        self,
+        map_handler: UDFMapCallable = None,
+        mapt_handler: UDFMapTCallable = None,
+        reduce_handler: UDFReduceCallable = None,
+        sock_path=MULTIPROC_FUNCTION_SOCK_PORT,
+        max_message_size=MAX_MESSAGE_SIZE,
+        max_threads=MAX_THREADS,
+    ):
+        if not (map_handler or mapt_handler or reduce_handler):
+            raise ValueError("Require a valid map/mapt handler and/or a valid reduce handler.")
+
+        self.__map_handler: UDFMapCallable = map_handler
+        self.__mapt_handler: UDFMapTCallable = mapt_handler
+        self.__reduce_handler: UDFReduceCallable = reduce_handler
+        self._max_message_size = max_message_size
+        self._max_threads = max_threads
+        self.cleanup_coroutines = []
+        # Collection for storing strong references to all running tasks.
+        # Event loop only keeps a weak reference, which can cause it to
+        # get lost during execution.
+        self.background_tasks = set()
+
+        self._server_options = [
+            ("grpc.max_send_message_length", self._max_message_size),
+            ("grpc.max_receive_message_length", self._max_message_size),
+        ]
+        self._sock_path = sock_path
+        self._PROCESS_COUNT = int(os.getenv("NUM_CPU_MULTIPROC", multiprocessing.cpu_count()))
+        self._THREAD_CONCURRENCY = self._PROCESS_COUNT
```

Comment:
Usually pep8 recommends instance variables to be always small case. Maybe `self.__process_count` and `self.__thread_concurrency`